packages/dialect-react-ui/components/Chat/provider/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, FunctionComponent, useContext } from 'react';
+
+// We are duplicating this type in order to have separation of concern: it may be different to what we have in props
+type ChatType = 'inbox' | 'popup' | 'vertical-slider';
+
+interface ChatContextValue {
+  id: string;
+  type: ChatType;
+  onChatClose?: () => void;
+  onChatOpen?: () => void;
+}
+
+interface ChatProviderProps {
+  id: string;
+  type: ChatType;
+  onChatClose?: () => void;
+  onChatOpen?: () => void;
+}
+
+export const ChatContext = createContext<ChatContextValue | null>(null);
```

Comment:
i guess it should be smth like
```suggestion
export const ChatContext = createContext<ChatContextValue>({} as ChatContextValue);
```